components/SearchAutocomplete.tsx

```diff
@@ -96,6 +103,47 @@ const SearchAutocomplete: FC<SearchAutocompleteProps> = ({
     options.length &&
     options.some(option => ('options' in option ? option.options.length : true))
 
+  const onKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
+    if (!options.length) {
+      return
+    }
+    if (!isOpen) {
+      onOpen()
+    }
+    let nextFocus = 0
+    const focusableOptions = options.reduce<SearchOption[]>((acc, option) => {
+      if ('options' in option) {
+        acc.push(...option.options)
+      } else {
+        acc.push(option)
+      }
+      return acc
+    }, [])
+    const focusedIndex = focusedOption
+      ? focusableOptions.indexOf(focusedOption)
+      : -1
```

Comment:
`indexOf` returns -1 by default, so I think you can always just do `focusableOptions.indexOf(focusedOption)` (e.g. `"cool".indexOf(null)` )